examples/seismic/acoustic/acoustic_example.py

```diff
@@ -52,16 +56,24 @@ def acoustic_setup(shape=(50, 50, 50), spacing=(15.0, 15.0, 15.0),
 
 def run(shape=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
         time_order=2, space_order=4, nbpml=40, full_run=False,
-        autotune=False, **kwargs):
+        autotune=False, constant=False, **kwargs):
 
     solver = acoustic_setup(shape=shape, spacing=spacing, nbpml=nbpml, tn=tn,
-                            space_order=space_order, time_order=time_order, **kwargs)
+                            space_order=space_order, time_order=time_order,
+                            constant=constant, **kwargs)
 
     initial_vp = smooth10(solver.model.m.data, solver.model.shape_domain)
     dm = np.float32(initial_vp**2 - solver.model.m.data)
     info("Applying Forward")
     rec, u, summary = solver.forward(save=full_run, autotune=autotune)
 
+    if constant:
+        # With  a new m as Constant
+        m0 = Constant(name="m", value=.25, dtype=np.float32)
+        solver.forward(save=full_run, m=m0)
+        # With a new m as a scalar value
+        solver.forward(save=full_run, m=.25)
```

Comment:
Should the non-constant code path still be executed as well, or should this go into an `else` clause here?

Reply:
>Should the non-constant code path still be executed as well

This is not a non-constant path.
The rest (adjoint, born and gradient) is generated for constant/non-constant according to m, this `if` only adds an extra "test" for the constant case that is done all over the place for the non-constant case making sure we runtime args works.

Reply:
Ok